### PR TITLE
lower opname cardinality on http tracing

### PR DESCRIPTION
**Pull request by cdaguerre**

### CODE REVIEW

File: src/Semantics/OperationName/ClientRequestOperationNameResolver.php [hunk 1]
```diff
@@ -13,6 +13,13 @@ class ClientRequestOperationNameResolver implements ClientRequestOperationNameRe
 {
     public function getOperationName(string $method, string $url, string $peerName): string
     {
+        $hostname = parse_url($url, \PHP_URL_HOST);
+        $protocol = parse_url($url, \PHP_URL_SCHEME);
+
+        if (!empty($hostname) && !empty($protocol)) {
+            $url = sprintf('%s://%s', $protocol, $hostname);
```
Comment (andrew-demb) [inline src/Semantics/OperationName/ClientRequestOperationNameResolver.php:20]: Is it mean that the operation name won't include URI at all, just ahost and a protocol?

Comment (cdaguerre) [reply src/Semantics/OperationName/ClientRequestOperationNameResolver.php:20]: Yes, indeed, you're right.
It should be added as attribute using semantic conventions, notably:

File: src/Semantics/Attribute/ClientRequestAttributeProvider.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+namespace Instrumentation\Semantics\Attribute;
+
+use OpenTelemetry\SemConv\TraceAttributes;
+
+class ClientRequestAttributeProvider implements ClientRequestAttributeProviderInterface
+{
+    /**
+     * @param array<string> $capturedHeaders
+     */
+    public function __construct(private array $capturedHeaders = [])
```
Comment (cyve) [inline src/Semantics/Attribute/ClientRequestAttributeProvider.php:19]: Should we allow to capture the request body as well ?

**Review by cyve — APPROVED** [review]

File: src/Semantics/Attribute/ClientRequestAttributeProvider.php (hunk 2 above)
Comment (cdaguerre) [reply src/Semantics/Attribute/ClientRequestAttributeProvider.php:19]: i dont think so... do you see a use case?

**Pull request merged**